pkg/cri/server/sandbox_run.go

```diff
@@ -310,6 +311,14 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 			return nil, fmt.Errorf("failed to setup network for sandbox %q: %w", id, err)
 		}
 
+		// Update metadata here to save CNI result and pod IPS to disk.
+		if err := container.Update(ctx,
+			// Update sandbox metadata to include NetNS info
+			containerd.UpdateContainerOpts(containerd.WithContainerExtension(sandboxMetadataExtension, &sandbox.Metadata)),
```

Comment:
would you please extend test case for this? thanks https://github.com/containerd/containerd/blob/dd5605e444cd7f3b599da40795268dc5f7f7f2e6/integration/sandbox_run_rollback_test.go#L143

Reply:
Sure

Reply:
I added a section on TestContainerdRestart that tests this after containerd has restarted

Reply:
I suggest let's verify the IPs are present in these test cases of sandbox_run_rollback_test . Basically the `sbStatus.Network.Ip` value.

1. https://github.com/containerd/containerd/blob/3bc8fc4d3067c32d2580e716af095a837c0fbe9a/integration/sandbox_run_rollback_test.go#L158
2. https://github.com/containerd/containerd/blob/3bc8fc4d3067c32d2580e716af095a837c0fbe9a/integration/sandbox_run_rollback_test.go#L223
3. https://github.com/containerd/containerd/blob/3bc8fc4d3067c32d2580e716af095a837c0fbe9a/integration/sandbox_run_rollback_test.go#L294